src/biz/collab/ops.rs

```diff
@@ -293,10 +303,86 @@ pub async fn get_user_trash_folder_views(
   Ok(section_items_to_trash_folder_view(&section_items, &folder))
 }
 
+#[allow(clippy::too_many_arguments)]
+fn patch_old_workspace_folder(
+  user: RealtimeUser,
+  workspace_id: &str,
+  folder: &mut Folder,
+  child_view_id_without_space: &[String],
+) -> Result<Vec<u8>, AppError> {
+  let encoded_update = {
+    let space_id = Uuid::new_v4().to_string();
+    let view = NestedChildViewBuilder::new(user.uid, workspace_id.to_string())
+      .with_view_id(space_id.clone())
+      .with_name("General")
+      .with_extra(|builder| {
+        let mut extra = builder.is_space(true, SpacePermission::PublicToAll).build();
+        extra["space_icon_color"] = json!("0xFFA34AFD");
+        extra["space_icon"] = json!("interface_essential/home-3");
+        extra
+      })
+      .build()
+      .view;
+    let mut txn = folder.collab.transact_mut();
+    folder.body.views.insert(&mut txn, view, None);
+    for view_id in child_view_id_without_space {
+      folder
+        .body
+        .move_nested_view(&mut txn, view_id, &space_id, None);
+    }
```

Comment:
```rs
    // associate the child with its new parent and place it after the prev_view_id. If the prev_view_id is None,
    // place it as the first child.
    self
      .views
      .associate_parent_child_with_txn(txn, new_parent_id, view_id, prev_view_id.clone());
```

I'm not sure if the order is handled correctly. The order should be preserved after the move.